sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -0,0 +1,630 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+
+#include "net/ipv6/addr.h"
+#include "net/gnrc/nettype.h"
+#include "net/gnrc/ipv6/netif.h"
+#include "net/gnrc/ipv6/nib.h"
+#include "net/gnrc/ndp2.h"
+#include "net/gnrc/pktqueue.h"
+#include "net/gnrc/sixlowpan/nd.h"
+#include "net/ndp.h"
+#include "net/sixlowpan/nd.h"
+
+#include "_nib-internal.h"
+#include "_nib-arsm.h"
+#include "_nib-6ln.h"
+#include "_nib-6lr.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+#if ENABLE_DEBUG
+#include "xtimer.h"
+#endif
+
+#if ENABLE_DEBUG
+static char addr_str[IPV6_ADDR_MAX_STR_LEN];
+#endif
+
+#if GNRC_IPV6_NIB_CONF_QUEUE_PKT
+static gnrc_pktqueue_t _queue_pool[GNRC_IPV6_NIB_NUMOF];
+#endif
+
+/**
+ * @internal
+ * @{
+ */
+static void _handle_nbr_sol(kernel_pid_t iface, const ipv6_hdr_t *ipv6,
+                            const ndp_nbr_sol_t *nbr_sol, size_t icmpv6_len);
+static void _handle_nbr_adv(kernel_pid_t iface, const ipv6_hdr_t *ipv6,
+                            const ndp_nbr_adv_t *nbr_adv, size_t icmpv6_len);
+
+static bool _resolve_addr(const ipv6_addr_t *dst, kernel_pid_t iface,
+                          gnrc_pktsnip_t *pkt, gnrc_ipv6_nib_nc_t *nce,
+                          _nib_onl_entry_t *entry);
+
+static void _handle_snd_na(gnrc_pktsnip_t *pkt);
+
+/* interface flag checks */
+#if GNRC_IPV6_NIB_CONF_ROUTER
+static inline bool _is_rtr(const gnrc_ipv6_netif_t *netif)
+{
+    return (netif->flags & GNRC_IPV6_NETIF_FLAGS_ROUTER);
+}
+#endif
+/** @} */
+
+void gnrc_ipv6_nib_init(void)
+{
+    evtimer_event_t *tmp;
+
+    mutex_lock(&_nib_mutex);
+    for (evtimer_event_t *ptr = _nib_evtimer.events;
+         (ptr != NULL) && (tmp = (ptr->next), 1);
+         ptr = tmp) {
+        evtimer_del((evtimer_t *)(&_nib_evtimer), ptr);
+    }
+    _nib_init();
+    mutex_unlock(&_nib_mutex);
+}
+
+void gnrc_ipv6_nib_init_iface(kernel_pid_t iface)
+{
+    _nib_iface_t *nib_iface;
+
+    assert(iface > KERNEL_PID_UNDEF);
+    DEBUG("nib: Initialize interface %u\n", (unsigned)iface);
+    mutex_lock(&_nib_mutex);
+    nib_iface = _nib_iface_get(iface);
+    assert(nib_iface != NULL);
+    /* TODO:
```

Comment:
question: should we put this into doxygen to make this TODO more obviouvs?

Reply:
These are implemented in #7456 and #7479, so why?